File: admin-dashboard/src/utils/orderStatus.js
export const ORDER_STATUSES = ['pending', 'preparing_shipping', 'ready_to_ship', 'shipped', 'delivered', 'cancelled']

export const ORDER_STATUS_LABELS = {
  pending: 'قيد الانتظار',
  preparing_shipping: 'قيد التجهيز',
  ready_to_ship: 'تم التجهيز',
  shipped: 'الشحن',
  delivered: 'تم التسليم',
  cancelled: 'ملغي',
}

export const ORDER_STATUS_COLORS = {
  pending: 'warning',
  preparing_shipping: 'primary',
  ready_to_ship: 'info',
  shipped: 'secondary',
  delivered: 'success',
  cancelled: 'error',
}

const LEGACY_STATUS_MAP = {
  confirmed: 'preparing_shipping',
  processing: 'preparing_shipping',
}

export function normalizeOrderStatus(status) {
  return LEGACY_STATUS_MAP[status] || status
}

export function getOrderStatusLabel(status) {
  const key = normalizeOrderStatus(status)
  return ORDER_STATUS_LABELS[key] || status
}

export function getOrderStatusColor(status) {
  const key = normalizeOrderStatus(status)
  return ORDER_STATUS_COLORS[key] || 'default'
}

File: fulfillment_app/lib/utils/order_status.dart
import 'package:flutter/material.dart';
import '../core/theme.dart';

class OrderStatusMeta {
  final String label;
  final String shortLabel;
  final Color color;
  final Color bg;
  final IconData icon;

  const OrderStatusMeta({
    required this.label,
    required this.shortLabel,
    required this.color,
    required this.bg,
    required this.icon,
  });
}

OrderStatusMeta orderStatusMeta(String status) {
  switch (status) {
    case 'pending':
      return const OrderStatusMeta(
        label: 'قيد الانتظار',
        shortLabel: 'انتظار',
        color: AppTheme.warning,
        bg: AppTheme.warningSoft,
        icon: Icons.hourglass_top_rounded,
      );
    case 'preparing_shipping':
      return const OrderStatusMeta(
        label: 'قيد التجهيز',
        shortLabel: 'تجهيز',
        color: AppTheme.info,
        bg: AppTheme.infoSoft,
        icon: Icons.inventory_2_rounded,
      );
    case 'shipped':
      return const OrderStatusMeta(
        label: 'قيد الشحن',
        shortLabel: 'شحن',
        color: Color(0xFF7C3AED),
        bg: Color(0xFFF3E8FF),
        icon: Icons.local_shipping_rounded,
      );
    case 'delivered':
      return const OrderStatusMeta(
        label: 'تم التسليم',
        shortLabel: 'مُسلّم',
        color: AppTheme.success,
        bg: AppTheme.successSoft,
        icon: Icons.check_circle_rounded,
      );
    case 'cancelled':
      return const OrderStatusMeta(
        label: 'ملغي',
        shortLabel: 'ملغي',
        color: AppTheme.danger,
        bg: AppTheme.dangerSoft,
        icon: Icons.cancel_rounded,
      );
    default:
      return const OrderStatusMeta(
        label: 'غير معروف',
        shortLabel: '—',
        color: AppTheme.textMuted,
        bg: AppTheme.surfaceAlt,
        icon: Icons.help_outline_rounded,
      );
  }
}

const orderFlowSteps = [
  'pending',
  'preparing_shipping',
  'shipped',
  'delivered',
];
